security-command-center/snippets/src/test/java/management/api/EventThreatDetectionCustomModuleTest.java

```diff
@@ -0,0 +1,166 @@
+  private static final int MAX_ATTEMPT_COUNT = 3;
+  private static final int INITIAL_BACKOFF_MILLIS = 120000; // 2 minutes
+
+  @Rule
+  public final MultipleAttemptsRule multipleAttemptsRule =
+      new MultipleAttemptsRule(MAX_ATTEMPT_COUNT, INITIAL_BACKOFF_MILLIS);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ID");
+  }
+
+  @AfterClass
+  public static void cleanUp() throws IOException {
+    // Perform cleanup after running tests
+    cleanupExistingCustomModules();
+  }
+
+  // cleanupExistingCustomModules clean up all the existing custom module
+  private static void cleanupExistingCustomModules() throws IOException {
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+      ListEventThreatDetectionCustomModulesRequest request =
+          ListEventThreatDetectionCustomModulesRequest.newBuilder()
+              .setParent(String.format("projects/%s/locations/global", PROJECT_ID))
+              .build();
+      ListEventThreatDetectionCustomModulesPagedResponse response =
+          client.listEventThreatDetectionCustomModules(request);
+      // Iterate over the response and delete custom module one by one which start with
+      // java_sample_custom_module
+      for (EventThreatDetectionCustomModule module : response.iterateAll()) {
+        if (module.getDisplayName().startsWith("java_sample_custom_module")) {
+          String customModuleId = extractCustomModuleId(module.getName());
+          deleteCustomModule(PROJECT_ID, customModuleId);
+        }
+      }
+    }
+  }
+
+  // extractCustomModuleID extracts the custom module Id from the full name
+  private static String extractCustomModuleId(String customModuleFullName) {
+    if (!Strings.isNullOrEmpty(customModuleFullName)) {
+      Pattern pattern = Pattern.compile(".*/([^/]+)$");
+      Matcher matcher = pattern.matcher(customModuleFullName);
+      if (matcher.find()) {
+        return matcher.group(1);
+      }
+    }
+    return "";
+  }
+
+  // createCustomModule method is for creating the custom module
+  private static EventThreatDetectionCustomModule createCustomModule(
+      String projectId, String customModuleDisplayName) throws IOException {
+    if (!Strings.isNullOrEmpty(projectId) && !Strings.isNullOrEmpty(customModuleDisplayName)) {
+      EventThreatDetectionCustomModule response =
+          CreateEventThreatDetectionCustomModule.createEventThreatDetectionCustomModule(
+              projectId, customModuleDisplayName);
+      return response;
+    }
+    return null;
+  }
```

Comment:
this method is redundant. it is an exact replica of createEventThreatDetectionCustomModule() except for testing for empty string arguments which look unnecessary for private method inside test class.

Reply:
Actually I have created this method because in every test case I am creating the custom module first and then performing the test case. So in order to use the same code I have created this method and calling it in every test case and also you might have noticed I have called the same method createEventThreatDetectionCustomModule() inside this method.

Reply:
Addressed, removed this separate method, now making the direct call in every testcase